src/device/libpixy_m4/inc/sdmmc.h

```diff
@@ -16,22 +16,35 @@
 #ifndef __SDMMC_H__
 #define __SDMMC_H__
 
-#include <stdint.h>
+#include "blobs.h"
+#include "pixytypes.h"
 
 #define SDMMC_HEADER_MAGIC    "MTTR"
 #define SDMMC_HEADER_VERSION  1
 
 typedef struct __attribute__((packed))
 {
-    uint32_t magic;
-    uint32_t version;
-    uint32_t boot_cnt;
-    uint16_t crc16;
+    uint32_t magic;               // A specific number to make sure the sd card is formatted correctly
+    uint32_t version;             // Version of header structure
+    uint32_t session_cnt;         // Current recording session
+    uint8_t  crc8;                // Cyclic Redundancy Check
 } SdmmcHeader;
 
+typedef struct __attribute__((packed))
+{
+    uint32_t session_cnt;         // Reference to session counter
+    uint32_t frame_cnt;           // Frame counter
+    uint32_t timestamp_us;        // Monotonic timestamp (microseconds since boot)
+    uint32_t last_write_time_us;  // Elapsed time of last SD Card write
```

Comment:
Elapsed time since last SD card write? Or timestamp of last write to SD?

Reply:
Time of last write. We can't afford to have more than one write per frame. The overhead is really high.